duckduckgo/Android
# Feature/periodic configuration downloading

Asana Issue URL: https://app.asana.com/0/471012189846020/481414989525314

## Description
Separation of tracker downloading and data loading, along with the scheduling of fetching updates for tracker data as well as HTTPS upgrade lists.

Makes of use [JobScheduler](https://developer.android.com/reference/android/app/job/JobScheduler.html) to handle the recurring background fetching of data. Currently set to approximately once every 24 hours, though this is deliberately inexact - Android will run the job when it thinks it best.

## Steps to Test this PR:
1. Uninstall existing app (breaking change to DB schema)
1. Ensure that data is downloaded when app first launches 
    - This can be confirmed by viewing using a Proxy, or by watching for the logcat message "Running immediate attempt to download app configuration"
1. Ensure that a job is scheduled with JobScheduler
    - This can be confirmed by using `adb shell dumpsys jobscheduler`
        - This will print output of existing jobs. Look for `com.duckduckgo.app.job.AppConfigurationJobService` in this list.
    - You can also force a job to run 
        - `adb shell cmd jobscheduler run -f com.duckduckgo.app 1`
        - requires Android 7.1 or greater to run this
1. ⚠️  Really important step - check tracker and HTTPs upgrade functionality still works!
